File: frontend/src/types/favorite.ts
export interface FavoriteItem {
  job_id: string
  title: string
  category: string
  company_name: string
  location: string | null
  favorited_at: string
}

export interface FavoriteSummary {
  company_id: string
  company_name: string
  count: number
}

File: backend/app/schemas/favorite.py
from pydantic import BaseModel


class FavoriteCreate(BaseModel):
    job_id: str


class FavoriteOut(BaseModel):
    job_id: str
    favorited_at: str


class FavoriteJobOut(BaseModel):
    job_id: str
    title: str
    category: str
    company_name: str
    location: list[str] = []
    favorited_at: str


class FavoriteSummaryItem(BaseModel):
    company_id: str
    company_name: str
    count: int
